hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/menu/DashboardMenu.java

```diff
@@ -177,34 +187,54 @@ private VerticalLayout buildLinksAndVersion() {
         return links;
     }
 
-    private static Resource getImage() {
+    private static Resource getImage(final boolean gravatar) {
+        if (!gravatar) {
+            return new ThemeResource("images/profile-pic-57px.jpg");
+        }
+
         return UserDetailsFormatter.getCurrentUserEmail().map(email -> (Resource) new GravatarResource(email))
                 .orElse(new ThemeResource("images/profile-pic-57px.jpg"));
 
     }
 
-    private static Component buildUserMenu() {
+    private Component buildUserMenu(final UiProperties uiProperties) {
         final MenuBar settings = new MenuBar();
         settings.addStyleName("user-menu");
         settings.setHtmlContentAllowed(true);
 
-        final MenuItem settingsItem = settings.addItem("", getImage(), null);
+        final MenuItem settingsItem = settings.addItem("", getImage(uiProperties.isGravatar()), null);
 
         final String formattedTenant = UserDetailsFormatter.formatCurrentTenant();
-        final String formattedUsername = UserDetailsFormatter.formatCurrentUsername();
-        String tenantAndUsernameHtml = "";
         if (!StringUtils.isEmpty(formattedTenant)) {
-            tenantAndUsernameHtml += formattedTenant + "<br>";
+            settingsItem.setText(formattedTenant);
+            settingsItem.setDescription(i18n.getMessage("menu.user.description",
+                    UserDetailsFormatter.getCurrentTenant(), UserDetailsFormatter.getCurrentUser().getUsername()));
+        } else {
+            settingsItem.setText("...");
         }
-        tenantAndUsernameHtml += formattedUsername;
-        settingsItem.setText(tenantAndUsernameHtml);
-        settingsItem.setDescription(formattedUsername);
+
         settingsItem.setStyleName("user-menuitem");
 
-        settingsItem.addItem("Sign Out", selectedItem -> Page.getCurrent().setLocation("/UI/logout"));
+        final String logoutUrl = generateLogoutUrl();
+
+        settingsItem.addItem("Sign Out", selectedItem -> Page.getCurrent().setLocation(logoutUrl));
         return settings;
     }
 
+    private static String generateLogoutUrl() {
+        try {
+            return UriComponentsBuilder.fromPath(LOGOUT_BASE)
+                    .queryParam("login",
+                            URLEncoder.encode(UriComponentsBuilder.fromPath(LOGIN_BASE)
+                                    .queryParam("tenant", UserDetailsFormatter.getCurrentTenant()).build().toString(),
+                                    StandardCharsets.UTF_8.toString()))
+                    .build().toString();
```

Comment:
Please have a look at https://docs.spring.io/spring/docs/current/javadoc-api/org/springframework/web/util/UriComponentsBuilder.html#toUriString-- if this call is sufficient for both URI creations (build, encode and toString)